pkgs/extension_discovery/lib/extension_discovery.dart

```diff
@@ -277,13 +273,19 @@ Future<List<Extension>> _findExtensions({
 
   return UnmodifiableListView(
     registry
-        .where((e) => e.present)
-        .map((e) => Extension._(
-              package: e.package,
-              rootUri: packageConfigUri.resolveUri(e.rootUri),
-              packageUri: e.packageUri,
-              config: e.config,
-            ))
+        .map((e) {
+          final config = e.config;
+          if (config == null) {
+            return null;
+          }
```

Comment:
instead of including null entries in the list, should we keep the where clause? 
`.where((e) => e.config != null)`

Reply:
Then we end up having a null-check assertion, but I don't really care too much :D